src/Sylius/Component/Core/Order/Requirements/RequiredBillingAddressSpecification.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Core\Order\Requirements;
+
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Specification\CompositeSpecification;
+
+class RequiredBillingAddressSpecification extends CompositeSpecification
+{
+    public function isSatisfiedBy(object $candidate): bool
+    {
+        if (false === $candidate instanceof OrderInterface) {
```

Comment:
```suggestion
        if ($candidate instanceof OrderInterface) {
```
isn't enough ?

Reply:
No. That changes the `instanceof` logic.

Reply:
true :D so
```suggestion
        if (!$candidate instanceof OrderInterface) {
```

Reply:
I personally don't like negation on instanceof - this is one of the PHP language-specific flaw.. but yea changed it for sake of codestyle

Reply:
Even though `instanceof` has precedence over `!`, some devs prefer to add parentheses "just in case". I'm pretty sure even I do it sometimes. 🤷‍♂️